DEVEL.md

```diff
@@ -1,7 +1,253 @@
+
+## Day-to-day Development How to's
+### Get the Source Code
+
+With `git` installed on the system, just clone the repository:
+
+```bash
+$ export INTEL_DEVICE_PLUGINS_SRC=/path/to/intel-device-plugins-for-kubernetes
+$ git clone https://github.com/intel/intel-device-plugins-for-kubernetes ${INTEL_DEVICE_PLUGINS_SRC}
+```
+
+### Build and Run Plugin Binaries
+
+With `go` development environment installed on the system, build the plugin:
+
+```bash
+$ cd ${INTEL_DEVICE_PLUGINS_SRC}
+$ make gpu_plugin
+```
+
+**Note:** All the available plugin build targets is roughly the output of `ls cmd`.
+
+To test the plugin binary on the development system, run as administrator:
+
+```bash
+$ sudo -E ${INTEL_DEVICE_PLUGINS_SRC}/cmd/dsa_plugin/gpu_plugin
+```
+
+### Build Container Images
+
+The dockerfiles are generated on the fly from `.in` suffixed files and `.docker` include-snippets which are stitched together with
+cpp preprocessor. You need to install cpp for that, e.g. in ubuntu it is found from build-essential (sudo apt install build-essential).
+Don't edit the generated dockerfiles. Edit the inputs.
+
+The simplest way to build all the docker images, is:
+```
+$ make images
+```
+
+But it is very slow. You can drastically speed it up by first running once:
+```
+$ make vendor
+```
+
+Which brings the libraries into the builder container without downloading them again and again for each plugin.
+
+But it is still slow. You can further speed it up by first running once:
+```
+$ make licenses
+```
+
+Which pre-creates the go-licenses for all plugins, instead of re-creating them for each built plugin, every time.
+
+But it is still rather slow to build all the images, and unnecessary, if you iterate on just one. Instead, build just the one you are iterating on, example:
+
+```
+$ make intel-gpu-plugin
+```
+
+**Note:** All the available image build targets is roughly the output of `ls build/docker/*.Dockerfile`.
+
+If you iterate on only one plugin and if you know what its target cmd is (see folder `cmd/`), you can opt to pre-create just its licenses, example:
+```
+$ make licenses/gpu_plugin
+```
+
+The container image target names in the Makefile are derived from the `.Dockerfile.in` suffixed filenames under folder `build/docker/templates/`.
+
+Recap:
+```
+$ make vendor
+$ make licenses (or just make licenses/gpu_plugin)
+$ make intel-gpu-plugin
+```
+
+Repeat the last step only, unless you change library dependencies. If you pull in new sources, start again from `make vendor`.
+
+**Note:** The image build tool can be changed from the default `docker` by setting the `BUILDER` argument
+to the [`Makefile`](Makefile): `make intel-gpu-plugin BUILDER=buildah`.
+
+### Build Against a Newer Version of Kubernetes
+
+First you need to update module dependencies. The easiest way is to use the
+script copied from https://github.com/kubernetes/kubernetes/issues/79384#issuecomment-521493597:
```

Comment:
should we have this script in the scripts/ directory of the repo?

Reply:
@bart0sh I thought about but decided to keep it as it is now. Can you help and take it in a separate PR and move us to `v1.25.0` APIs at the same time?

Reply:
done in a separate PR as suggested: https://github.com/intel/intel-device-plugins-for-kubernetes/pull/1131